### PR TITLE
Lib 8 create book catalog web

### PR DESCRIPTION
**Pull request by Nadavrav**

### CODE REVIEW

File: libraroo-web/src/App.tsx [hunk 1]
```diff
@@ -12,7 +13,7 @@ function App() {
         <Route path="/" element={<ProtectedRoute><TextField/></ProtectedRoute>} />
         <Route path="/register" element={<Register />} />
         <Route path="/login" element={<Login />} />
-
+        <Route path="/AddBook" element={<CreateBookCatalog />} />
```
Comment (IdoRoss) [inline libraroo-web/src/App.tsx:16]: path="book-catalog/add-book"

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        language,
+      });
+
+      navigate("/");
+    } catch (err: any) {
+      setError("Invalid email or password");
+    } finally {
+      setLoading(false);
+    }
+  };
```
Comment (IdoRoss) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:33]: wtf

**Review by IdoRoss — CHANGES_REQUESTED** [review]
Also make sure you created lang validation taks in jira

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx
Comment (zikili) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:1]: Maybe change the language <TextField> to <select> between multiple language code options

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          disabled={loading}
+        >
+          Add book
+        </Button>
+      </Box>
+    </Container>
+  );
+};
+
+export default CreateBookCatalog;
```
Comment (IdoRoss) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:97]: Remove default

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+import BookCatalogService from "../services/bookCatalog.service";
```
Comment (IdoRoss) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:3]: Change to bookCatalogService as this is a instance of a class

File: libraroo-web/src/App.tsx (hunk 1 above)
Comment (Nadavrav) [reply libraroo-web/src/App.tsx:16]: yes okay

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx (hunk 2 above)
Comment (Nadavrav) [reply libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:33]: oopsooshhhhhhhhh

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx (hunk 4 above)
Comment (Nadavrav) [reply libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:3]: yes okay yes

**Review by zikili — CHANGES_REQUESTED** [review]

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx (hunk 2 above)
Comment (IdoRoss) [reply libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:33]: I dont think this should navigate you to the home screen, maybe the book catalog list screen, add a TODO comment

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+          required
+          fullWidth
+          id="language"
+          label="Language"
+          name="language"
+          autoComplete="language"
+          autoFocus
+          value={language}
+          onChange={(e) => setLanguage(e.target.value)}
+        />
```
[the remark this answers is not in the document]
Comment (IdoRoss) [reply libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:81]: Nad should add a future task for lang dropdown in jira according to https://en.wikipedia.org/wiki/List_of_ISO_639_language_codes

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx (hunk 3 above)
Comment (IdoRoss) [reply libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:97]: Cause when exposing a component/function (which arent instances) you want to expose it explicitly.
Right now you can do:
```import dompfil from ".../CreateBookCatalog";```

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    try {
+      await bookCatalogService .addBookCatalog({
+        author,
+        title,
+        language,
+      });
+
+      // TODO : Navigate to book catalog list screen.
+    } catch (err: any) {
+      setError("An error occured while trying to add book, check and try again.");
```
Comment (IdoRoss) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:29]: First check if this is an axios error, then print the message from the API.
if not print this mesage

**Review by IdoRoss — CHANGES_REQUESTED** [review]

File: libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+        } else {
+          // An error occurred while setting up the request
+          console.log("Error during request setup:", err.message);
+          setError("An unexpected error occurred. Please try again.");
+        }
+      } else {
+        // Handle non-Axios errors
+        console.log("Unexpected error:", err);
+        setError("Unexpected error occurred.");
+      }
```
Comment (IdoRoss) [inline libraroo-web/src/bookCatalog/components/CreateBookCatalog.tsx:48]: Move logic to a utils func "getApiErrorMessage" also remove irrelevant stuff like status codes

**Review by IdoRoss — CHANGES_REQUESTED** [review]

File: libraroo-web/src/shared/utils.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
+import { AxiosInstance, isAxiosError } from "axios";
+import apiClient from "./api-client";
```
Comment (IdoRoss) [inline libraroo-web/src/shared/utils.ts:2]: remove

File: libraroo-web/src/shared/utils.ts [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+export class Utils {
+
+
+  constructor(private readonly apiClient: AxiosInstance) {}
```
Comment (IdoRoss) [inline libraroo-web/src/shared/utils.ts:7]: remove

File: libraroo-web/src/shared/utils.ts [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+
+  static async getErrorMessage(err: any) {
+    var message;
```
Comment (IdoRoss) [inline libraroo-web/src/shared/utils.ts:10]: let message: string = "";

File: libraroo-web/src/shared/utils.ts [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+        console.log("No response received:", err.request);
+        message = "Network error: No response from server.";
+      } else {
+        // An error occurred while setting up the request
+        console.log("Error during request setup:", err.message);
+        message = "An unexpected error occurred. Please try again.";
+      }
+    } else {
+      // Handle non-Axios errors
+      console.log("Unexpected error:", err);
```
Comment (IdoRoss) [inline libraroo-web/src/shared/utils.ts:34]: Remove all console.logs

**Review by IdoRoss — CHANGES_REQUESTED** [review]

**Review by IdoRoss — APPROVED** [review]

**Review by zikili — APPROVED** [review]

**Pull request merged**